backend/common/utils/rollup.py

```diff
@@ -46,15 +46,21 @@ def find_descendants_per_cell_type(cell_types):
     descendants_per_cell_type : list
         List of lists of descendants for each cell type in the input list.
     """
-
+    lookup_table = {}
+    cell_types_set = set(cell_types)
     relatives_per_cell_type = []
     for cell_type in cell_types:
-        relatives = descendants(cell_type)
-        relatives_per_cell_type.append(relatives)
-
-    for i, children in enumerate(relatives_per_cell_type):
-        # remove descendent cell types that are not cell types in the WMG data
-        relatives_per_cell_type[i] = list(set(children).intersection(cell_types))
+        if ";;" in cell_type:
```

Comment:
Cell types can be suffixed with ";;" if descendants need to be calculated within groups only.

For example, `"CL:0000540;;UBERON:0002048--PATO:0000461"` means that CL:0000540 is in tissue UBERON:0002048 and disease PATO:0000461. Descendants of CL:0000540 will be restricted to those that are in the above group.